milli/src/update/index_documents/mod.rs

```diff
@@ -326,6 +326,16 @@ impl<'t, 'u, 'i, 'a> IndexDocuments<'t, 'u, 'i, 'a> {
         R: io::Read,
         F: Fn(UpdateIndexingStep, u64) + Sync,
     {
+        let mut reader = BufReader::new(reader);
+        reader.fill_buf()?;
+
+        // Early return when there is no document to add
+        if reader.buffer().is_empty() {
+            return Ok(DocumentAdditionResult {
+                nb_documents: 0,
+            })
+        }
+
```

Comment:
Is it a way to make sure that we do not create a default primary key that is "id" because no documents (and therefore no primary key) were found?

Reply:
No, it's a way to not throw a "could not infer primary key" on an empty addition 🙂

Reply:
Or is it? 🤔 I'm tired, maybe I understood @irevoire issue wrong

Reply:
Yeah that was my issue

Reply:
Tell me if this solves it ☺️ (next week)

Reply:
I have a client side fix for this, but I am not sure whose duty it is to do that, milli's or the clien?

Reply:
But doesn't this commit fix the issue, here?

Reply:
yes but I ended up implementing a fix in dumps so we could merge it. We can leave it here, and 'll remove it from dumps. The question is, should you get an error for pushing an empty document addition or not? I claim that we shouldn't, but I wanted to know if you thought otherwise.

Reply:
You must not get an error if you push an empty update, that's valid, even for the first update (where we guess the primary key).